test_integration/test_ee_short_region_slop.py

```diff
@@ -325,11 +364,135 @@ def extract_data_region(report, cols, key):
         sample = []
         for (rn, rr) in regions:
              try:
-                 sample.append(report.get_field(rr, cc))
+                 sample.append(raw_report.get_field(rr, cc))
              except KeyError:
                  sample.append(None)
         result[cc] = sample
     return pandas.DataFrame(result)
 
+
+def generate_report_plot(report_df, region_type, xaxis, yaxis, ylim=None):
+    for policy_type in ('fixed', 'dynamic'):
+        prof_name = 'ee_short_region_slop-{}'.format(policy_type)
+        level = ('profile-name', 'region-name')
+        key = (prof_name, region_type)
+        selected_data = report_df.xs(key=key, level=level).groupby('count')
+        xdata = selected_data.mean()[xaxis]
+        ydata = selected_data.mean()[yaxis]
+        ydata_std = selected_data.std()[yaxis]
+        plt.errorbar(xdata, ydata, ydata_std, fmt='.')
+        if ylim is not None:
+            plt.ylim(ylim)
+    plt.title('{} region {}'.format(region_type, yaxis))
+    plt.xlabel(xaxis)
+    plt.ylabel(yaxis)
+    plt.legend(('fixed', 'dynamic'))
+
+
+def read_trace(path):
+    converters={'REGION_HASH': lambda xx: int(xx, 16)}
+    return pandas.read_csv(path, sep='|', comment='#', converters=converters)
+
+
+def get_ipc(trace):
+    di = trace['INSTRUCTIONS_RETIRED'].diff()
+    dc = trace['CYCLES_THREAD'].diff()
+    return di / dc
+
+def get_freq(trace):
+    return trace['FREQUENCY']
+
+def get_relative_time(trace):
+    return trace['TIME'] - trace['TIME'].iloc[0]
+
+
+def get_uncore_freq(trace):
+    return trace['MSR::UNCORE_PERF_STATUS:FREQ']
+
+
+def select_region_count(trace, region_count):
+    return trace.loc[trace['REGION_COUNT'] == region_count]
+
+
+def select_region(trace, region_prefix, extension=''):
+    """e.g.:
+    >>> select_region(trace, 'scaling_2')
+    >>> select_region(trace, 'scaling_', range(12))
+    >>> select_region(trace, '', ['scaling_3', 'timed_3'])
+
+    """
+    if type(extension) is str:
+        extension = [extension]
+    region_hash = []
+    for ext in extension:
+        region_name = '{}{}'.format(region_prefix, ext)
+        region_hash.append(geopmpy.hash.crc32_str(region_name))
+    return trace.loc[trace['REGION_HASH'].isin(region_hash)]
+
+
+def select_time_range(trace, duration, begin_time=None):
```

Comment:
I like these helper funcs.  I'm wondering how we should organize a #import with all of them.

Reply:
Seems like a good idea for another PR.